src/os/bluestore/BlueStore.cc

```diff
@@ -4447,13 +4447,13 @@ void BlueStore::MempoolThread::_update_cache_settings()
 #define dout_prefix *_dout << "bluestore.OmapIteratorImpl(" << this << ") "
 
 BlueStore::OmapIteratorImpl::OmapIteratorImpl(
-  PerfCounters* _logger, CollectionRef c, OnodeRef& o, KeyValueDB::Iterator it)
+  PerfCounters* _logger, CollectionRef c, OnodeRef& o, KeyValueDB::Iterator it, string start_from)
   : logger(_logger), c(c), o(o), it(it)
 {
   logger->inc(l_bluestore_omap_iterator_count);
   std::shared_lock l(c->lock);
   if (o->onode.has_omap()) {
-    o->get_omap_key(string(), &head);
+    o->get_omap_key(start_from, &head);
```

Comment:
`head` is a key that can contain special value, "omap header", see `ObjectStore::omap_get_header()`.
Setting it to `start_from` would break `OmapIterator::seek_to_first()`.

Instead, we should `it->lower_bound(start_from)`,

Reply:
Oops, fixed the dissection of `bounds.lower_bound`.